components/OpenChallengeCard.js

```diff
@@ -0,0 +1,96 @@
+import * as WebBrowser from "expo-web-browser";
+import React from "react";
+import FlipComponent from "react-native-flip-component";
+import ChallengeCard from "./ChallengeCard";
+import RevealChallenge from "./RevealChallenge";
+import { StyleSheet, View } from "react-native";
+import timer from "react-native-timer";
+import { millisecondsToString } from "../utils/time";
+
+export default class OpenChallengeCard extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isFlipped: false,
+      timeLeftInMilliseconds: this.props.timeLeftInMilliseconds
+    };
+    this.flip = this.flip.bind(this);
+  }
+
+  componentDidMount() {
+    timer.setTimeout(
+      this,
+      this.props.assignedChallengeId,
+      () => {
+        // TODO: Kristie replace with modal
+        // Also decide what to do if time is up
+        alert("Time's up!");
+      },
+      this.props.timeLeftInMilliseconds
+    );
+    timer.setInterval(
+      this,
+      this.props.assignedChallengeId,
+      () => {
+        this.setState({
+          timeLeftInMilliseconds: this.state.timeLeftInMilliseconds - 1000
+        });
+      },
+      1000
+    );
+  }
+
+  componentWillUnmount() {
+    timer.clearTimeout(this);
+    timer.clearInterval(this);
+  }
+
+  flip = () => {
+    this.setState({
+      isFlipped: !this.state.isFlipped
```

Comment:
The boolean is necessary because it controls the isFlipped prop of the FlipComponent. We could change !this.state.isFlipped to true since it only goes one way but doesn't really matter.